src/v/kafka/server/handlers/find_coordinator.cc

```diff
@@ -14,157 +14,357 @@
 #include "config/configuration.h"
 #include "kafka/protocol/errors.h"
 #include "kafka/protocol/find_coordinator.h"
+#include "kafka/protocol/types.h"
 #include "kafka/server/coordinator_ntp_mapper.h"
+#include "kafka/server/request_context.h"
 #include "kafka/server/rm_group_frontend.h"
 #include "model/metadata.h"
+#include "security/acl.h"
 
 #include <seastar/util/log.hh>
 
 #include <fmt/ostream.h>
 
-#include <string_view>
+#include <exception>
+#include <expected>
+#include <iterator>
 
 namespace kafka {
 
-static ss::future<response_ptr>
-handle_leader(request_context& ctx, model::node_id leader) {
+// these are both fine so long as no future's execution outlives its
+// enclosing scope, here all coroutines are immediately awaited
+// NOLINTBEGIN(cppcoreguidelines-avoid-reference-coroutine-parameters)
+// NOLINTBEGIN(cppcoreguidelines-avoid-capturing-lambda-coroutines)
+namespace {
+
+ss::logger logger("find_coordinator_handler");
```

Comment:
please do not add per handler loggers, they may be fine for debugging but we are exposing them in REST API